lib/spack/spack/build_environment.py

```diff
@@ -581,7 +581,8 @@ def setup_package(pkg, dirty):
 
     # traverse in postorder so package can use vars from its dependencies
     spec = pkg.spec
-    for dspec in pkg.spec.traverse(order='post', root=False, deptype='build'):
+    for dspec in pkg.spec.traverse(order='post', root=False,
+                                   deptype=('build', 'test')):
```

Comment:
@scheibelp @tgamblin  What is the semantic of `link`-type deps? Asking as I expected to find also `link` in the deptypes above - even though practically speaking we probably have no `link`-type dep that is not also `build`-type.  Does it make sense adding it?